libs/core/allocator_support/include/hpx/allocator_support/thread_local_caching_allocator.hpp

```diff
@@ -82,8 +83,9 @@ namespace hpx::util {
                 }
                 else
                 {
-                    p = data.top().first;
-                    data.pop();
+                    std::pair<T*, size_type> pair;
+                    data.pop(pair);
```

Comment:
I think you should check for the return value here as between the check `data.empty()` above and here somebody could have removed the item. Even better if you rewrote the whole functions to something like:
```
std::pair<T*, size_type> pair;
if (!data.pop(pair)) {
    // allocate new
}
else {
   p = pair.first;
}
```

Reply:
Fixed, thanks!